packages/client-core/src/common/services/RouterService.tsx

```diff
@@ -87,7 +87,7 @@ export const useSearchParamState = () => {
   /** Create a nested reactor to react to state changes */
   useEffect(() => {
     const SubReactor = (props: { keyID: string }) => {
-      const value = useHookstate(getMutableState(SearchParamState)[props.keyID]).value
+      const value = getMutableState(SearchParamState).value[props.keyID]
```

Comment:
why the change from useHookstate to getMutableState here? this means it will not trigger reactive updates to the react context

Reply:
I can't remember why I changed that, and it was still working when I reverted it, so I've undone this.